File: server/src/Timers/MoveProjectilesTimer.cs
using SpacetimeDB;
using Game = pillz.server.Reducers.Game;

namespace pillz.server.Timers;

[Table(Scheduled = nameof(Game.MoveProjectiles), ScheduledAt = nameof(ScheduledAt))]
public partial struct MoveProjectilesTimer
{
    [PrimaryKey, AutoInc]
    public ulong ScheduledId;
    public ScheduleAt ScheduledAt;
}

File: server-rust/src/timers/move_projectiles_timer.rs
use spacetimedb::ScheduleAt;
use crate::reducers::game::move_projectiles;

#[spacetimedb::table(name = move_projectiles_timer, scheduled(move_projectiles))]
pub struct MoveProjectilesTimer {
    #[primary_key]
    #[auto_inc]
    pub scheduled_id: u64,

    pub scheduled_at: ScheduleAt,
}
